cpp/util/Histogram.h

```diff
@@ -420,15 +495,10 @@ template<typename T> class Histogram
     template<typename ComputeFunction>
     void reduceOverThreadsPerBin(ThreadLocalHistogram& local_histograms, const ComputeFunction& cf)
     {
+        local_histograms.reduceHistogramsInto(m_bin_counts);
         util::forLoopWrapper(0, m_bin_counts.size(), [=](size_t begin, size_t end) {
             for (size_t i = begin; i < end; ++i)
             {
-                for (typename ThreadLocalHistogram::const_iterator local_bins = local_histograms.begin();
-                     local_bins != local_histograms.end(); ++local_bins)
-                {
-                    m_bin_counts[i] += (*local_bins).m_bin_counts[i];
-                }
-
                 cf(i);
```

Comment:
⚠️ This assumes that the order of reducing bin counts and calling cf doesn't matter.

Reply:
Actually I'd frame this as a design choice, not an assumption; the docstring for the function explicitly states that it's for post-processing. I can see how it might get misused if a user isn't careful, but I don't see any point in addressing the pre-processing case unless there's a need since this version covers all existing use-cases in freud.

Reply:
Ok, I'll resolve this.